From 690f2f35563a1f7a89b796f9b0e7996627dbda21 Mon Sep 17 00:00:00 2001
From: Robin Gong <B38343@freescale.com>
Date: Fri, 18 Nov 2011 10:52:32 +0800
Subject: [PATCH] ENGR00162578 DMA mx5: increase DMA Zone size to 112

Increase DMA zone size from 96 to 112 size, and default size is 112,
change imx5_defconfig, change SPBA0_BASE_ADDR_VIRT from 0xFB100000 to
0xF7C00000 , so that it can't overlap with DMA zone
Signed-off-by: Robin Gong <B38343@freescale.com>
(cherry picked from commit 2d04dcb9b717a7c46358987f41a03141eccc42b0)
---
 arch/arm/configs/imx5_defconfig         |    7 ++++---
 arch/arm/plat-mxc/Kconfig               |    4 ++--
 arch/arm/plat-mxc/include/mach/memory.h |    4 ++--
 arch/arm/plat-mxc/include/mach/mx5x.h   |    2 +-
 4 files changed, 9 insertions(+), 8 deletions(-)

diff --git a/arch/arm/configs/imx5_defconfig b/arch/arm/configs/imx5_defconfig
index 419adde..3a0cc96 100644
--- a/arch/arm/configs/imx5_defconfig
+++ b/arch/arm/configs/imx5_defconfig
@@ -1,7 +1,7 @@
 #
 # Automatically generated make config: don't edit
 # Linux kernel version: 2.6.35.3
-# Wed Jun  1 20:11:44 2011
+# Fri Nov 18 10:24:37 2011
 #
 CONFIG_ARM=y
 CONFIG_HAVE_PWM=y
@@ -266,7 +266,7 @@ CONFIG_ARCH_MXC_HAS_NFC_V3=y
 CONFIG_ARCH_MXC_HAS_NFC_V3_2=y
 CONFIG_MXC_BLUETOOTH_RFKILL=y
 CONFIG_IRAM_ALLOC=y
-CONFIG_DMA_ZONE_SIZE=96
+CONFIG_DMA_ZONE_SIZE=112
 CONFIG_ISP1504_MXC=y
 CONFIG_UTMI_MXC=y
 # CONFIG_MXC_IRQ_PRIOR is not set
@@ -1230,9 +1230,9 @@ CONFIG_SENSORS_MAX17135=y
 # CONFIG_SENSORS_W83627EHF is not set
 # CONFIG_SENSORS_LIS3_SPI is not set
 # CONFIG_SENSORS_LIS3_I2C is not set
+CONFIG_SENSORS_IMX_AHCI=y
 CONFIG_SENSORS_MAG3110=y
 CONFIG_SENSORS_ISL29003=y
-CONFIG_SENSORS_IMX_AHCI=y
 CONFIG_MXC_MMA8450=y
 CONFIG_MXC_MMA8451=y
 # CONFIG_THERMAL is not set
@@ -2112,6 +2112,7 @@ CONFIG_MXC_MC13892_POWER=y
 CONFIG_MXC_MC34708_ADC=y
 CONFIG_MXC_MC34708_RTC=y
 CONFIG_MXC_MC34708_BATTERY=m
+# CONFIG_MXC_MC34708_PWM is not set
 # CONFIG_MXC_PMIC_MC9S08DZ60 is not set
 
 #
diff --git a/arch/arm/plat-mxc/Kconfig b/arch/arm/plat-mxc/Kconfig
index 926379f..7a4412a 100644
--- a/arch/arm/plat-mxc/Kconfig
+++ b/arch/arm/plat-mxc/Kconfig
@@ -87,8 +87,8 @@ config MXC_FB_IRAM
 
 config DMA_ZONE_SIZE
 	int "DMA memory zone size"
-	range 0 96
-	default 24
+	range 0 112
+	default 112
 	help
 	  This is the size in MB for the DMA zone. The DMA zone is used for
 	  dedicated memory for large contiguous video buffers
diff --git a/arch/arm/plat-mxc/include/mach/memory.h b/arch/arm/plat-mxc/include/mach/memory.h
index 83532f6..2e66516 100644
--- a/arch/arm/plat-mxc/include/mach/memory.h
+++ b/arch/arm/plat-mxc/include/mach/memory.h
@@ -1,5 +1,5 @@
 /*
- * Copyright (C) 2004-2010 Freescale Semiconductor, Inc. All Rights Reserved.
+ * Copyright (C) 2004-2011 Freescale Semiconductor, Inc. All Rights Reserved.
  */
 
 /*
@@ -63,7 +63,7 @@
 #else
 
 #ifdef CONFIG_ARCH_MX5
-#define CONSISTENT_DMA_SIZE	(96 * SZ_1M)
+#define CONSISTENT_DMA_SIZE	(112 * SZ_1M)
 #else
 #define CONSISTENT_DMA_SIZE	(32 * SZ_1M)
 #endif
diff --git a/arch/arm/plat-mxc/include/mach/mx5x.h b/arch/arm/plat-mxc/include/mach/mx5x.h
index 5c16710..96f6686 100644
--- a/arch/arm/plat-mxc/include/mach/mx5x.h
+++ b/arch/arm/plat-mxc/include/mach/mx5x.h
@@ -194,7 +194,7 @@
  * SPBA global module enabled #0
  */
 #define SPBA0_BASE_ADDR 	0x70000000
-#define SPBA0_BASE_ADDR_VIRT	0xFB100000
+#define SPBA0_BASE_ADDR_VIRT	0xF7C00000
 #define SPBA0_SIZE		SZ_1M
 
 #define MMC_SDHC1_BASE_ADDR	(SPBA0_BASE_ADDR + 0x00004000)
-- 
1.5.4.4

